src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1169,241 @@ const isMobileFieldUpdateAllowed = (
+  }
+
+  logger.info({
+    message: `no msgSrvcName, creating Twilio credentials for form ${formId}`,
+    meta: {
+      action: 'createTwilioCredentials',
+      formId: formId,
+      msgSrvcName,
+    },
+  })
+
+  return ResultAsync.fromPromise(
+    FormModel.startSession().then((session: ClientSession) =>
+      session
+        .withTransaction(
+          async () =>
+            await createTwilioTransaction(formId, msgSrvcName, body, session),
+        )
+        .then(() => session.endSession()),
+    ),
+    (error) => {
+      logger.error({
+        message: 'Error encountered when creating Twilio Secret',
+        meta: {
+          action: 'createTwilio',
+          body,
+        },
+        error,
+      })
+
+      return new SecretsManagerError('Error occurred when creating Twilio!')
+    },
+  )
+}
+
+export const createTwilioTransaction = async (
+  formId: string,
+  msgSrvcName: string,
+  body: CreateSecretRequest,
+  session: ClientSession,
+): Promise<void> => {
+  const doc = await FormModel.updateMsgSrvcName(formId, msgSrvcName)
+
+  if (doc) {
+    await secretsManager.createSecret(body).promise()
+    await doc.save({ session })
+  }
+}
+
+export const updateTwilioCredentials = (
+  msgSrvcName: string,
+  twilioCredentials: TwilioCredentials,
+): ResultAsync<
+  number,
+  SecretsManagerError | SecretsManagerNotFoundError | TwilioCacheError
+> => {
+  const twilioCredentialsData: TwilioCredentialsData =
+    new TwilioCredentialsData(twilioCredentials)
+
+  const body: PutSecretValueRequest = {
+    SecretId: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+  }
+
+  return ResultAsync.fromPromise(
+    secretsManager.getSecretValue({ SecretId: msgSrvcName }).promise(),
+    (error) => {
+      logger.error({
+        message: 'Twilio Credentials do not exist in Secrets Manager',
+        meta: {
+          action: 'updateTwilioCredentials',
+          msgSrvcName,
+        },
+        error,
+      })
+
+      return new SecretsManagerNotFoundError(
+        'Twilio Credentials do not exist in Secrets Manager',
+      )
+    },
+  )
+    .andThen(() => {
+      logger.info({
+        message: 'Twilio Credentials has been found in Secrets Manager',
+        meta: {
+          action: 'updateTwilioCredentials',
+          msgSrvcName,
+        },
+      })
+
+      return ResultAsync.fromPromise(
+        secretsManager.putSecretValue(body).promise(),
+        (error) => {
+          logger.error({
+            message: 'Error occurred when updating Twilio in Secret Manager!',
+            meta: {
+              action: 'updateTwilioCredentials',
+              body,
+            },
+            error,
+          })
+
+          return new SecretsManagerError(
+            'Error occurred when updating Twilio in Secret Manager!',
+          )
+        },
+      )
+    })
+    .andThen(() => {
+      return ResultAsync.fromPromise(
+        Promise.resolve(twilioClientCache.del(msgSrvcName)),
```

Comment:
I don't think you need to wrap this in a fromPromise, etc since from its docs "A delete will never fail."

We can just return like so:

```ts
.map(() => twilioClientCache.del(msgSrvcName))
```